pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -1117,6 +1145,26 @@ spec:
 			"Normal Started",
 			"Warning Failed PipelineRun foo/pipeline-mismatching-param-type parameters have mismatching types",
 		},
+	}, {
+		name: "invalid-pipeline-missing-object-keys",
+		pipelineRun: parse.MustParsePipelineRun(t, `
+metadata:
+  name: pipeline-missing-object-param-keys
+  namespace: foo
+spec:
+  pipelineRef:
+    name: a-pipeline-with-object-params
+  params:
+    - name: some-param
+      value:
+        key1: "a"
+`),
+		reason:         ReasonObjectParameterMissKeys,
+		permanentError: true,
+		wantEvents: []string{
+			"Normal Started",
+			"Warning Failed PipelineRun foo/pipeline-missing-object-param-keys parameters is missing some object keys",
```

Comment:
This error message would be more helpful if it returned which keys were missing and for which parameters

Reply:
sg! Changed the verifier helper function to return a map for param<->missing keys so when the users see the err message from pipelinerun status, they will see which keys were missing and for which parameters.

But for the wantEvent, if I also put the returned err message here, it gives weird error and fails test though the expected and got event reported from the following message are exactly same. I guess this is why other wantEvent cuts off the second half message too.

```
logger.go:130: 2022-06-15T11:43:08.897-0700     ERROR   TestReconcile_InvalidPipelineRuns/invalid-pipeline-missing-object-keys  pipelinerun/reconciler.go:294Returned an error       {"targetMethod": "ReconcileKind", "error": "1 error occurred:\n\t* PipelineRun missing object keys for parameters: map[some-param:[key2]]\n\n"}
        pipelinerun_test.go:1316: error in test pipeline-missing-object-param-keys: expected event "Warning Failed PipelineRun foo/pipeline-missing-object-param-keys parameters is missing object keys required by Pipeline foo/a-pipeline-with-object-params's parameters: PipelineRun missing object keys for parameters: map[some-param:[key2]]" but got "Warning Failed PipelineRun foo/pipeline-missing-object-param-keys parameters is missing object keys required by Pipeline foo/a-pipeline-with-object-params's parameters: PipelineRun missing object keys for parameters: map[some-param:[key2]]" instead
```

Reply:
That's really weird-- can you try it out on the cluster and see what error message is returned to the user?

Reply:
sure. Just added.

Reply:
This is the error I refer to when we also add the rest of the error message to the expected `wantEvent`.
<img width="1907" alt="Screen Shot 2022-06-17 at 17 51 37" src="https://user-images.githubusercontent.com/30398455/174405648-b285d18e-8254-4c33-8ca0-7c880b77ae2a.png">

So I guess we need to remove the rest part to pass the unit test @lbernick

Reply:
But in the error returned when users miss some keys, we can see the full message without a problem with the information about which keys were missing and for which parameters.